### PR TITLE
Add ChildProcessPromiseResult type to ChildProcess

### PR DESCRIPTION
**Pull request by johnsonw**
Signed-off-by: Will Johnson <william.c1.johnson@gmail.com>

### CODE REVIEW

File: fable/ChildProcess.fs [hunk 1]
```diff
@@ -6,13 +6,15 @@ namespace Fable.Import.Node.PowerPack
...
 module ChildProcess =
+    open Fable.Import
     open Fable.Import.Node
     open Fable.Core.JsInterop
     open Fable.Core
     open Fable.PowerPack
 
     type Stdout = Stdout of string
     type Stderr = Stderr of string
+    type ChildProcessPromiseResult = JS.Promise<Result<Stdout * Stderr, ChildProcess.ExecError * Stdout * Stderr>>
```
Comment (jgrund) [inline fable/ChildProcess.fs]: You should build up the enclosed types as well (minus the result), as I imagine we will want to refer to those as well.

Comment (johnsonw) [reply fable/ChildProcess.fs]: Sure, I'll separate it out to `ChildProcessResultOk` and `ChildProcessResultError`.

Comment (jgrund) [reply fable/ChildProcess.fs]: Maybe just `Out` and `Err`?

Comment (johnsonw) [reply fable/ChildProcess.fs]: Oh just saw this, i'll rename.

File: fable/ChildProcess.fs [hunk 2]
```diff
@@ -6,13 +6,17 @@ namespace Fable.Import.Node.PowerPack
...
     open Fable.Import.Node
     open Fable.Core.JsInterop
     open Fable.Core
     open Fable.PowerPack
 
     type Stdout = Stdout of string
     type Stderr = Stderr of string
+    type Out = Stdout * Stderr
+    type Err = ChildProcess.ExecError * Stdout * Stderr
+    type ChildProcessPromiseResult = JS.Promise<Result<Out, Err>>
```
Comment (jgrund) [inline fable/ChildProcess.fs]: Perhaps it's worth using this type to annotate the return, so we know it stays in sync?

Comment (johnsonw) [reply fable/ChildProcess.fs]: Sure, couldn't hurt.

**Pull request merged**